aten/src/ATen/gen.py

```diff
@@ -175,7 +175,7 @@ def check_all_files_written(self):
 }
 
 backends = ['CPU', 'CUDA']
-densities = ['Dense', 'Sparse']
+densities = ['Dense', 'Sparse', 'Mkldnn']  # TODO: layout instead of densities?
```

Comment:
Yes, layouts sounds much better. Also, let's capitalize MKLDNN for consistency with CPU and CUDA capitalization.

Reply:
@ezyang Oh, since MKL-DNN is a layout instead of a device, I named it "Mkldnn" to follow the naming convention of layouts, e.g. "Dense" and "Sparse", not capitalizing it like "CPU" or "CUDA".